src/ConnectedMode/Migration/Wizard/MigrationWizardWindow.xaml.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Windows;
+using Microsoft.VisualStudio.PlatformUI;
+using System.ComponentModel;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Migration.Wizard
+{
+    public sealed partial class MigrationWizardWindow : DialogWindow
+    {
+        public event EventHandler StartMigration;
+
+        private bool dialogResult;
+
+        internal MigrationWizardWindow()
+        {
+            InitializeComponent();
```

Comment:
Technically I should set all the correct visibility states here (buttons and pages) to ensure that it is set up correctly. I chose against this for now because it is cluttering and considering we do not actually know how the ui will look like unnecessary.